src/Model/InvoiceManager.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Model;
+
+use PDO;
+
+class InvoiceManager extends AbstractManager
+{
+    public const TABLE2 = 'invoice';
+    public const TABLE3 = 'product';
+
+    /**
+     * Insert invoice in database
+     */
+    public function insert(array $invoice): void
+    {
+        // Insérer les données dans la table 'invoice'
+        $invoiceQuery = "INSERT INTO " . self::TABLE2 . "
+                 (total_amount, created_at, due_at, user_siret, user_name, user_address, user_bank_details,
+                 client_siret, client_name, client_address, user_id)
+                 VALUES
+                 (:total_amount, :created_at, :due_at, :user_siret, :user_name, :user_address, :user_bank_details,
+                 :client_siret, :client_name, :client_address, :user_id)";
+
+        $invoiceStatement = $this->pdo->prepare($invoiceQuery);
+
+        $invoiceStatement->bindValue(':total_amount', $invoice['total_amount'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':created_at', $invoice['created_at'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':due_at', $invoice['due_at'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':user_siret', $invoice['user_siret'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':user_name', $invoice['user_name'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':user_address', $invoice['user_address'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':user_bank_details', $invoice['user_bank_details'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':client_siret', $invoice['client_siret'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':client_name', $invoice['client_name'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':client_address', $invoice['client_address'], PDO::PARAM_STR);
+        $invoiceStatement->bindValue(':user_id', 1, PDO::PARAM_INT); // CODER EN DURE POUR LE MOMENT CHANGER PLUS TARD
+
+        $invoiceStatement->execute();
+
+        // Récupérer l'ID de la facture nouvellement insérée
+        $invoiceId = $this->pdo->lastInsertId();
+
+        $products = [];
+
+        // On boucle sur le tableau invoice
+        foreach($invoice as $key => $value) {
+            // On verifie les clés commencant par "product_name_"
+            if (strpos($key, 'product_name_') === 0){
+
+                // Extraire le numéro du produit
+                $product_number = substr($key, strlen('product_name_'));
```

Comment:
Il faudrait ici laisser la logique d'insertion des produits dans le controller. 

L'idée serait d'appeler, dans votre controller : `$this->invoiceManager->insert($data)`
puis une fois que c'est fait, toujours dans le controller, faire une boucle parcourant toutes les données des produits et pour chaque élément appeler une fonction `$this->productManager->insert($data)`

Reply:
Séparation des logiques d'insertion faites.